tests/accounting/test_simplepnl.py

```diff
@@ -9,174 +10,199 @@ def test_initial_state():
     assert pnl_tracker.current_ndx == 0, f"Expected current_ndx=0, got {pnl_tracker.current_ndx}"
     assert pnl_tracker.pending_decisions == [], f"Expected pending_decisions=[], got {pnl_tracker.pending_decisions}"
     assert pnl_tracker.pnl_data == [], f"Expected pnl_data=[], got {pnl_tracker.pnl_data}"
 
+
     # Step 2: Record first decision at step 0 (y = 100, decision = 1)
     pnl_tracker.tick(x=100, horizon=100, decision=1)
-    assert len(
-        pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
+    assert len(pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
 
     # Step 3: Move forward by 50 steps without resolving (y = 120, decision = 0)
     pnl_tracker.current_ndx = 50
     pnl_tracker.tick(x=120, horizon=50, decision=0)
-    assert len(
-        pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
+    assert len(pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
 
     # Step 4: Move forward another 50 steps and resolve first decision (y = 150)
     pnl_tracker.current_ndx = 100
     pnl_tracker.tick(x=150, horizon=100, decision=0)
-    assert len(
-        pnl_tracker.pending_decisions) == 0, f"Expected 0 pending decisions after resolution, got {len(pnl_tracker.pending_decisions)}"
+    assert len(pnl_tracker.pending_decisions) == 0, f"Expected 0 pending decisions after resolution, got {len(pnl_tracker.pending_decisions)}"
     assert len(pnl_tracker.pnl_data) == 1, f"Expected 1 PnL record, got {len(pnl_tracker.pnl_data)}"
-    expected_pnl = 150 - 100 - pnl_tracker.epsilon  # PnL = y_resolution - y_decision - epsilon
-    actual_pnl = pnl_tracker.pnl_data[0][-1]
+    expected_pnl = 150 - 100 - pnl_tracker.epsilon
+    actual_pnl = pnl_tracker.pnl_data[0]['pnl']
     assert actual_pnl == expected_pnl, f"Expected PnL={expected_pnl}, got {actual_pnl}"
 
     # Step 5: Record second decision at step 100 (y = 150, decision = -1)
     pnl_tracker.tick(x=150, horizon=100, decision=-1)
-    assert len(
-        pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
+    assert len(pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
 
     # Step 6: Move forward 50 steps without resolving (y = 140)
     pnl_tracker.current_ndx = 150
     pnl_tracker.tick(x=140, horizon=50, decision=0)
-    assert len(
-        pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
+    assert len(pnl_tracker.pending_decisions) == 1, f"Expected 1 pending decision, got {len(pnl_tracker.pending_decisions)}"
 
     # Step 7: Resolve second decision (y = 100)
-    pnl_tracker.current_ndx = 200
+    pnl_tracker.current_ndx = 201
```

Comment:
@microprediction This is the only change I am not sure about.

Reply:
Instead of frying my brain, I created a simpler test:

`test_periodic_decisions`

that is easier to check so I am somewhat confident this is ok.

Reply:
and the same with backoff. Often a bit easier to break apart functionalities.

Reply:
ok